boards/shared/Test/Src/Test_SharedStateMachine.cpp

```diff
@@ -0,0 +1,95 @@
+#include <gtest/gtest.h>
+#include <fff.h>
+
+extern "C"
+{
+#include "App_SharedStateMachine.h"
+}
+
+#include "App_TestWorld.h"
+
+FAKE_VOID_FUNC(state_A_entry, struct StateMachine *);
+FAKE_VOID_FUNC(state_A_tick_1kHz, struct StateMachine *);
+FAKE_VOID_FUNC(state_A_exit, struct StateMachine *);
+FAKE_VOID_FUNC(state_B_entry, struct StateMachine *);
+FAKE_VOID_FUNC(state_B_tick_1Hz, struct StateMachine *);
+FAKE_VOID_FUNC(state_B_tick_1kHz, struct StateMachine *);
+FAKE_VOID_FUNC(state_B_exit, struct StateMachine *);
+
+static struct State state_A;
+static struct State state_B;
+
+class SharedStateMachineTest : public testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        world = App_TestWorld_Create();
+
+        state_A.run_on_entry     = state_A_entry;
+        state_A.run_on_tick_1Hz  = state_A_tick_1Hz;
+        state_A.run_on_tick_1kHz = state_A_tick_1kHz;
+        state_A.run_on_exit      = state_A_exit;
+
+        state_B.run_on_entry     = state_B_entry;
+        state_B.run_on_tick_1Hz  = state_B_tick_1Hz;
+        state_B.run_on_tick_1kHz = state_B_tick_1kHz;
+        state_B.run_on_exit      = state_B_exit;
+
+        state_machine = App_SharedStateMachine_Create(world, &state_A);
+
+        RESET_FAKE(state_A_entry);
+        RESET_FAKE(state_A_tick_1kHz);
+        RESET_FAKE(state_A_exit);
+        RESET_FAKE(state_B_entry);
+        RESET_FAKE(state_B_tick_1Hz);
+        RESET_FAKE(state_B_tick_1kHz);
+        RESET_FAKE(state_B_exit);
+    }
+
+    void TearDown() override
+    {
+        ASSERT_TRUE(world != NULL);
+        App_TestWorld_Destroy(world);
+        world = NULL;
+
+        ASSERT_TRUE(state_machine != NULL);
+        App_SharedStateMachine_Destroy(state_machine);
+        state_machine = NULL;
```

Comment:
nit; Could start using the `TearDownObject` I created in #591

Reply:
Nice :smile: changed.